ghost/core/core/server/services/email-analytics/EmailAnalyticsServiceWrapper.js

```diff
@@ -100,13 +111,34 @@ class EmailAnalyticsServiceWrapper {
         }
         this.fetching = true;
 
+        // NOTE: Data shows we can process ~2500 events per minute on Pro for a large-ish db (150k members).
+        //       This can vary locally, but we should be conservative with the number of events we fetch.
         try {
-            const c1 = await this.fetchLatest({maxEvents: Infinity});
-            const c2 = await this.fetchMissing({maxEvents: Infinity});
-
-            // Only fetch scheduled if we didn't fetch a lot of normal events
-            await this.fetchScheduled({maxEvents: 20000 - c1 - c2});
-
+            // Prioritize opens since they are the most important (only data directly displayed to users)
+            const c1 = await this.fetchLatestOpenedEvents({maxEvents: 10000});
+            if (c1 >= 10000) {
```

Comment:
Will this ever be true if maxEvents is set to 10000?

Reply:
Also just out of curiosity, how did you arrive at 10,000 as the right number for this? 2500 events per minute * 4 minutes?

Reply:
I'm being a bit conservative. The thought was the restart the job's logic _at least_ every 5 mins, so 2500*4 gives a little head room as that 2500 is variable and has an inverse relationship with the db size (email_recips table size, primarily).

I'm not following the first comment. It's >= not just >. I saw some notes previously that `maxEvents` wasn't super precise. In testing I've been seeing it fairly precise. I was going to put it at 8000 or something but that seemed very arbitrary and difficult for someone reading this to understand. Not that it's easy as is.